src/Native/Unix/System.Security.Cryptography.Native/pal_asn1.cpp

```diff
@@ -67,24 +67,26 @@ extern "C" void CryptoNative_Asn1BitStringFree(ASN1_STRING* a)
     ASN1_BIT_STRING_free(a);
 }
 
-extern "C" ASN1_OCTET_STRING* CryptoNative_DecodeAsn1OctetString(const uint8_t* buf, int32_t len)
+extern "C" ASN1_OCTET_STRING* CryptoNative_Asn1OctetStringNew()
 {
-    if (!buf || !len)
+    ASN1_OCTET_STRING *r = ASN1_OCTET_STRING_new();
+    if (r == nullptr)
     {
-        return nullptr;
+        ERR_clear_error();
     }
 
-    return d2i_ASN1_OCTET_STRING(nullptr, &buf, len);
-}
-
-extern "C" ASN1_OCTET_STRING* CryptoNative_Asn1OctetStringNew()
-{
-    return ASN1_OCTET_STRING_new();
+    return r;
 }
 
 extern "C" int32_t CryptoNative_Asn1OctetStringSet(ASN1_OCTET_STRING* s, const uint8_t* data, int32_t len)
 {
-    return ASN1_OCTET_STRING_set(s, data, len);
+    int32_t r = ASN1_OCTET_STRING_set(s, data, len);
+    if (r == 0)
+    {
+        ERR_clear_error();
```

Comment:
Why do these two clear in native when everything else is clearing in managed?

Reply:
I tried to clean things in native when it was possible, i.e.: managed code really ignored the error and proceeded, but the downside is the lack of consistency...

Reply:
Just out of curiosity, how can you tell that this one can't be cleared in managed code?

Reply:
It can, but, on the managed flow the code proceeds despite of this failure and can succeed despite of the failure so it seemed reasonable simply to clear on the native side, the downside be a bit of inconsistency.

Reply:
That makes sense, thanks.